Can't "Launch Extension" for debugging

### Version Info
```
Version: 1.86.2
Commit: 903b1e9d8990623e3d7da1df3d33db3e42d80eda
Date: 2024-02-13T19:41:37.860Z
Electron: 27.2.3
ElectronBuildId: 26908389
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Linux x64 6.6.7-arch1-1
```

### Reproducible Steps
1. Complete [setup steps](https://github.com/jdneo/vscode-checkstyle/blob/9714346e78e288ca0ee98b4a02dd12f9cb6b5517/CONTRIBUTING.md#setup) 1-6
2. Skip step 7 because jls seems to have an issue, but I did "right click the target.target file and select Reload Target Platform", then restart VS Code.
3. Run "Launch Extension" from debug view.


### What I expect to happen vs. What's actually happening
**Expected**: A new VS Code window should open up with the extension running. Afterwards I should be able to debug.
**Actual**: The `"preLaunchTask": "npm: watch"` seems to complete successfully and seems to be serving the webpack. No window opens.

### Console logs
I'm not sure these logs have to do with the problem, but it's possible. The issue seems to be with jls. Everything seems to build correctly even though it says `Project 'com.shengchen.checkstyle.checker' is missing required Java project: 'com.shengchen.checkstyle.runner'`

```
console.ts:137 [Extension Host] rejected promise not handled within 1 second: Error: com.shengchen.checkstyle.runner does not exist
p @ console.ts:137
console.ts:137 [Extension Host] stack trace: Error: com.shengchen.checkstyle.runner does not exist
	at /home/iborrelli/.vscode/extensions/redhat.java-1.28.1-linux-x64/dist/extension.js:2:1104120
	at ie (/home/iborrelli/.vscode/extensions/redhat.java-1.28.1-linux-x64/dist/extension.js:2:1104414)
	at /home/iborrelli/.vscode/extensions/redhat.java-1.28.1-linux-x64/dist/extension.js:2:1099198
	at Immediate.<anonymous> (/home/iborrelli/.vscode/extensions/redhat.java-1.28.1-linux-x64/dist/extension.js:2:1099218)
	at process.processImmediate (node:internal/timers:476:21)
p @ console.ts:137
mainThreadExtensionService.ts:81 [redhat.java]com.shengchen.checkstyle.runner does not exist
$onExtensionRuntimeError @ mainThreadExtensionService.ts:81
mainThreadExtensionService.ts:82 Error: com.shengchen.checkstyle.runner does not exist
	at /home/iborrelli/.vscode/extensions/redhat.java-1.28.1-linux-x64/dist/extension.js:2:1104120
	at ie (/home/iborrelli/.vscode/extensions/redhat.java-1.28.1-linux-x64/dist/extension.js:2:1104414)
	at /home/iborrelli/.vscode/extensions/redhat.java-1.28.1-linux-x64/dist/extension.js:2:1099198
	at Immediate.<anonymous> (/home/iborrelli/.vscode/extensions/redhat.java-1.28.1-linux-x64/dist/extension.js:2:1099218)
	at process.processImmediate (node:internal/timers:476:21)
```

Thanks!

--- a/.gitignore
+++ b/.gitignore
@@ -27,11 +27,6 @@ hs_err_pid*
 # build
 target/
 
-# Eclipse
-.classpath
-.project
-.settings
-
 # Checkstyle
 **/.checkstyle
 

--- a/.vscode/extensions.json
+++ b/.vscode/extensions.json
@@ -2,7 +2,7 @@
 	// See http://go.microsoft.com/fwlink/?LinkId=827846
 	// for the documentation about the extensions.json format
 	"recommendations": [
-		"ms-vscode.vscode-typescript-tslint-plugin",
-		"yaozheng.vscode-pde"
+		"yaozheng.vscode-pde",
+		"amodio.tsl-problem-matcher"
 	]
 }

--- a/.vscode/tasks.json
+++ b/.vscode/tasks.json
@@ -4,22 +4,7 @@
         {
             "type": "npm",
             "script": "watch",
-            "problemMatcher": {
-                "owner": "typescript",
-                "pattern":[
-                    {
-                        "regexp": "\\[tsl\\] ERROR",
-                        "file": 1,
-                        "location": 2,
-                        "message": 3
-                    }
-                ],
-                "background": {
-                    "activeOnStart": true,
-                    "beginsPattern": "Compilation \\w+ starting…",
-                    "endsPattern": "Compilation\\s+finished"
-                }
-            },
+            "problemMatcher": ["$ts-webpack-watch", "$tslint-webpack-watch"],
             "isBackground": true,
             "presentation": {
                 "reveal": "never"

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -40,13 +40,14 @@ The extension has three major modules, which are listed as follow:
 
 ### Setup
 0. Make sure you have latest LTS JDK, Node.js, VS Code and [Java Extension Pack](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack) installed.
-1. Fork and clone the repository: `git clone https://github.com/jdneo/vscode-checkstyle.git`
-2. `cd vscode-checkstyle`
-3. Install the node dependencies: `npm install`
-4. Build the Java modules: `npm run build-plugin`
-6. Open the directory `vscode-checkstyle` in VS Code
-7. Install the [Eclipse PDE Support extension](https://marketplace.visualstudio.com/items?itemName=yaozheng.vscode-pde) in your VS Code
-8. Open a Java file and wait until 👍 shows in the right-bottom of the status bar
+1. Fork and clone the repository: `git clone https://github.com/jdneo/vscode-checkstyle.git`.
+2. `cd vscode-checkstyle`.
+3. Install the node dependencies: `npm install`.
+4. Build the Java modules: `npm run build-plugin`.
+6. Open the directory `vscode-checkstyle` in VS Code.
+7. Install the [Eclipse PDE Support extension](https://marketplace.visualstudio.com/items?itemName=yaozheng.vscode-pde) in your VS Code.
+8. Install the [TypeScript + Webpack Problem Matchers](https://marketplace.visualstudio.com/items?itemName=amodio.tsl-problem-matcher) in your VS Code.
+9. Open a Java file and wait until 👍 shows in the right-bottom of the status bar
     > Note: Sometimes, if you find the code navigation is not working in the Java code, please try: 
     > - right click the [target.target](https://github.com/jdneo/vscode-checkstyle/blob/master/jdtls.ext/com.shengchen.checkstyle.target/target.target) file and select `Reload Target Platform`.
     > - Reload your VS Code.

--- a/README.md
+++ b/README.md
@@ -18,7 +18,7 @@
 </p>
 
 ## Requirements
-- JDK (version 1.8.0 or later)
+- JDK (version 17 or later)
 - VS Code (version 1.30.0 or later)
 - [Language Support for Java by Red Hat](https://marketplace.visualstudio.com/items?itemName=redhat.java)
 

--- a/jdtls.ext/com.shengchen.checkstyle.checker/pom.xml
+++ b/jdtls.ext/com.shengchen.checkstyle.checker/pom.xml
@@ -34,7 +34,7 @@
         </plugins>
     </build>
     <properties>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
     </properties>
 </project>

--- a/jdtls.ext/com.shengchen.checkstyle.runner/.classpath
+++ b/jdtls.ext/com.shengchen.checkstyle.runner/.classpath
@@ -1,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17"/>
+	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
+	<classpathentry kind="src" path="src/main/java"/>
+	<classpathentry kind="output" path="target/classes"/>
+</classpath>

--- a/jdtls.ext/com.shengchen.checkstyle.runner/.project
+++ b/jdtls.ext/com.shengchen.checkstyle.runner/.project
@@ -1,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>com.shengchen.checkstyle.runner</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.ManifestBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.SchemaBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.m2e.core.maven2Builder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.m2e.core.maven2Nature</nature>
+		<nature>org.eclipse.pde.PluginNature</nature>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+	</natures>
+	<filteredResources>
+		<filter>
+			<id>1708997989680</id>
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
+				<arguments>node_modules|\.git|__CREATED_BY_JAVA_LANGUAGE_SERVER__</arguments>
+			</matcher>
+		</filter>
+	</filteredResources>
+</projectDescription>

--- a/jdtls.ext/com.shengchen.checkstyle.runner/.settings/org.eclipse.core.resources.prefs
+++ b/jdtls.ext/com.shengchen.checkstyle.runner/.settings/org.eclipse.core.resources.prefs
@@ -1,0 +1,2 @@
+eclipse.preferences.version=1
+encoding/<project>=UTF-8

--- a/jdtls.ext/com.shengchen.checkstyle.runner/.settings/org.eclipse.m2e.core.prefs
+++ b/jdtls.ext/com.shengchen.checkstyle.runner/.settings/org.eclipse.m2e.core.prefs
@@ -1,0 +1,4 @@
+activeProfiles=
+eclipse.preferences.version=1
+resolveWorkspaceProjects=true
+version=1

--- a/jdtls.ext/com.shengchen.checkstyle.runner/META-INF/MANIFEST.MF
+++ b/jdtls.ext/com.shengchen.checkstyle.runner/META-INF/MANIFEST.MF
@@ -4,7 +4,7 @@ Bundle-Name: com.shengchen.checkstyle.runner
 Bundle-SymbolicName: com.shengchen.checkstyle.runner;singleton:=true
 Bundle-Version: 1.4.2
 Bundle-Activator: com.shengchen.checkstyle.runner.CheckstylePlugin
-Bundle-RequiredExecutionEnvironment: JavaSE-1.8
+Bundle-RequiredExecutionEnvironment: JavaSE-17
 Import-Package: org.eclipse.jdt.core,
  org.eclipse.jdt.launching,
  org.osgi.framework;version="1.3.0"

